backend/Api/Projects/ProjectController.cs

```diff
@@ -39,9 +42,76 @@ public ProjectController(ApplicationContext context, IMemoryCache cache)
                         new EngagementReadModel(e.Id, e.Name, e.State, false)).ToList()))
             .ToList();
     }
+
+    [HttpPut]
+    public ActionResult<ProjectWithConsultantsReadModel> Put([FromRoute] string orgUrlKey,
+        [FromBody] EngagementWriteModel body)
+    {
+        var service = new StorageService(_cache, _context);
+
+        var selectedOrg = _context.Organization.SingleOrDefault(org => org.UrlKey == orgUrlKey);
+        if (selectedOrg is null) return BadRequest("Selected org not found");
+
+        var customer = service.UpdateOrCreateCustomer(selectedOrg, body.CustomerName, orgUrlKey);
+        var project = service.UpdateOrCreateProject(customer, body, orgUrlKey);
+
+        var consultants = _context.Consultant.Where(c => body.ConsultantIds.Contains(c.Id)).ToList();
+        if (!consultants.Any()) return BadRequest("No consultant-ids match in db");
+
+        // Adding consultants to project
+        // TODO : What if consultants exists from before on project?
+        project.Consultants.AddRange(consultants);
+
+        var nextWeekSet = FillProjectWeekWithConsultants(consultants, project);
+
+        _context.SaveChanges();
+        _cache.Remove($"{ConsultantCacheKey}/{orgUrlKey}");
+
+        var readModels = new ReadModelFactory(service).GetConsultantReadModelsForWeeks(orgUrlKey, nextWeekSet);
+
+        var filteredConsultants = readModels.Where(c => body.ConsultantIds.Contains(c.Id)).ToList();
+
+        var responseModel = new ProjectWithConsultantsReadModel(project.Name, customer.Name, project.State,
+            filteredConsultants, project.IsBillable);
+
+        return Ok(responseModel);
+    }
+
+    private List<Week> FillProjectWeekWithConsultants(List<Consultant> consultants, Project project)
+    {
+        var thisWeek = Week.FromDateTime(DateTime.Now);
+        var nextWeekSet = thisWeek.GetNextWeeks(8);
+
+        const double emptyHours = 37.5;
+
+        foreach (var consultant in consultants)
+        {
+            foreach (var week in nextWeekSet)
+            {
+                var staffing = _context.Staffing
+                    .FirstOrDefault(s => s.ProjectId.Equals(project.Id)
+                                         && s.ConsultantId.Equals(consultant.Id)
+                                         && s.Week.Equals(week));
+                if (staffing is null)
+                {
+                    _context.Add(new Staffing
+                    {
+                        ProjectId = project.Id,
+                        Project = project,
+                        ConsultantId = consultant.Id,
+                        Consultant = consultant,
+                        Week = week,
+                        Hours = emptyHours
+                    });
+                }
+                else
+                    staffing.Hours = emptyHours;
+            }
+        }
+
+        return nextWeekSet;
+    }
```

Comment:
Det jeg tror er problemet her, er at vi jobber rett på databaseobjektet, og i listen. Derfor er det veldig vanskelig å gjøre dette kompitabelt med StorageService sånn den er nå. 

På den ene siden er jo dette smooth nok det, men samtidig er det et lite antipattern ift. hva vi har gjort hittil med `CreateOrUpdate..`

Men jeg er ikke helt sikker på hvem som skal definere den flyten. Kanskje SS skal ha to metoder: `Create` og `Update`, så er det kontrolleren som får bestemme. 

Så jeg sier ikke dette er feil, men det utfordrer litt arkitekturen vi har pr nå. Vi kan jo kjenne litt etter hva som er best da

Reply:
Hmm. Leser jeg tankene riktig med at selve "write" operasjonene bør være mer distinkte her og isolerte? Altså en funksjon/method for Create og en for Update?

Det skal for øvrig settes 0 timer på emptyHours, men ønsker å se hvordan dette blir med StorModalen først, så tar jeg den justeringen etter hvert.

Reply:
Lol 😄 Snakk om å spørre om det du allerede har svart på (sier jeg til meg selv 🙃)

En detalj her jo at jeg har gjort det litt "grovt" med å bare sammenligne "name" fields på project og customer her og ikke bruke Id-felt. Noe ryddigere kode, men det er ingen garanti på unique som ID gir.

Da må jeg endre PK på project/customer, noe som kan være en løsning.

Reply:
Sånn ideelt sett burde kanskje `FillProjectWeekWithConsultants` ikke være nødvendig syns jeg
..